pkg/collector/runner/running_checks_tracker.go

```diff
@@ -0,0 +1,75 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package runner
+
+import (
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+)
+
+// RunningChecksTracker is an object that keeps a thread-safe track of
+// all the running checks
+type RunningChecksTracker struct {
+	runningChecks map[check.ID]check.Check // The list of checks running
+	accessLock    sync.Mutex               // To control races on runningChecks
+}
+
+// NewRunningChecksTracker is a contructor for a RunningChecksTracker
+func NewRunningChecksTracker() *RunningChecksTracker {
+	return &RunningChecksTracker{
+		runningChecks: make(map[check.ID]check.Check),
+	}
+}
+
+// Check returns a check in the running check list, if it can be found
+func (t *RunningChecksTracker) Check(id check.ID) (check.Check, bool) {
+	t.accessLock.Lock()
+	defer t.accessLock.Unlock()
+
+	check, found := t.runningChecks[id]
+	return check, found
+}
+
+// AddCheck adds a check to the list of running checks if the check
+// isn't already added. Method returns a boolean if the addition was
+// successful.
+func (t *RunningChecksTracker) AddCheck(check check.Check) bool {
+	t.accessLock.Lock()
+	defer t.accessLock.Unlock()
+
+	if _, found := t.runningChecks[check.ID()]; found {
+		return false
+	}
+
+	t.runningChecks[check.ID()] = check
+	return true
+}
+
+// DeleteCheck removes a check from the list of running checks
+func (t *RunningChecksTracker) DeleteCheck(id check.ID) {
+	t.accessLock.Lock()
+	defer t.accessLock.Unlock()
+
+	delete(t.runningChecks, id)
+}
+
+// RunningChecks returns a list of all the running checks
+func (t *RunningChecksTracker) RunningChecks() map[check.ID]check.Check {
+	// TODO: Copy the list?
```

Comment:
Closures/anonymous functions aren't a really common pattern/standard for Golang in my experience but in this case it would make sense to do it that way. 👍